core/src/main/java/org/mule/runtime/core/registry/MuleRegistryHelper.java

```diff
@@ -181,6 +182,24 @@ public Transformer lookupTransformer(DataType source, DataType result) throws Tr
     protected Transformer resolveTransformer(DataType source, DataType result) throws TransformerException
     {
         Lock readLock = transformerResolversLock.readLock();
+        //if (transformerResolvers.isEmpty())
```

Comment:
remove. Ugly bug you found though. I wonder how many WTF moments we had because of this